tools/oam-converter/pkg/app/confdata.go

```
@@ -74,202 +61,149 @@ func ConfData() error {
 	if err != nil {
 		return err
 
 	}
+	//write the nonOAM resources to the file
 	return nil
 }
 
-// writeKubeResources Write the kube resources to the files in output directory
-func writeKubeResources(outputDirectory string, outputResources *types.KubeResources) (err error) {
-
-	//Write virtual services to files
-	if outputResources.VirtualServices != nil {
-		for _, virtualService := range outputResources.VirtualServices {
-
-			fileName := "" + virtualService.Name + ".yaml"
-			filePath := filepath.Join(outputDirectory, fileName)
-
-			f, err := os.Create(filePath)
-			if err != nil {
-				return err
-			}
-
-			defer f.Close()
-
-			r, err := json.Marshal(virtualService)
-			if err != nil {
-				return err
-			}
-			output, err := yaml.JSONToYAML(r)
-			if err != nil {
-				return err
-			}
-
-			_, err = f.WriteString(string(output))
-			if err != nil {
-				return err
-			}
-
-		}
-	}
-
-	//Write down gateway to files
-	if outputResources.Gateway != nil {
-
-		fileName := "gateway.yaml"
-		filePath := filepath.Join(outputDirectory, fileName)
-
-		f, err := os.Create(filePath)
+// Write the kube resources to the files in output directory
+func writeKubeResources(outputDirectory string, outputResources []unstructured.Unstructured) error {
+	for index := range outputResources {
+		err := writeToDirectory(outputDirectory, outputResources[index])
 		if err != nil {
 			return err
 		}
-		defer f.Close()
-		gatewayYaml, err := yaml.Marshal(outputResources.Gateway)
-		if err != nil {
-			return fmt.Errorf("failed to marshal YAML: %w", err)
-		}
+	}
+	return nil
+}
 
-		_, err = f.WriteString(string(gatewayYaml))
+func convertAnytoUnstructured(anyResources []any) []unstructured.Unstructured {
+	var unstructuredResources []unstructured.Unstructured
+	for _, obj := range anyResources {
+		resource, err := ToUnstructured(obj)
 		if err != nil {
-			return fmt.Errorf("failed to write YAML to file: %w", err)
+			print("Null obj")
```

should return an error here